core/src/main/scala/org/apache/spark/scheduler/TaskSetManager.scala

```diff
@@ -87,7 +87,13 @@ private[spark] class TaskSetManager(
   // number of slots on a single executor, would the task manager speculative run the tasks if
   // their duration is longer than the given threshold. In this way, we wouldn't speculate too
   // aggressively but still handle basic cases.
-  val speculationTasksLessEqToSlots = numTasks <= (conf.get(EXECUTOR_CORES) / sched.CPUS_PER_TASK)
+  // SPARK-30417: #cores per executor might not be set in spark conf for standalone mode, then
+  // the value of the conf would 1 by default. However, the executor would use all the cores on
+  // the worker. Therefore, CPUS_PER_TASK is okay to be greater than 1 without setting #cores.
+  // To handle this case, we assume the minimum number of slots is 1.
```

Comment:
Will this case fails this check:

https://github.com/apache/spark/blob/e1ea806b3075d279b5f08a29fe4c1ad6d3c4191a/core/src/main/scala/org/apache/spark/SparkContext.scala#L2737

It seems `checkResourcesPerTask` also doesn't handle this case properly. So, `CPUS_PER_TASK` greater than 1 should fail the check.

Reply:
I'm fixing that issue with https://issues.apache.org/jira/browse/SPARK-30446